aries/aries_vcx_core/src/wallet2/mod.rs

```diff
@@ -0,0 +1,137 @@
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use derive_builder::Builder;
+#[cfg(feature = "vdrtools_wallet")]
+use indy_api_types::domain::wallet::Record as IndyRecord;
+use serde::{Deserialize, Serialize};
+
+use crate::errors::error::VcxCoreResult;
+
+#[cfg(feature = "vdrtools_wallet")]
+pub mod indy_wallet;
+
+pub enum SigType {
+    EdDSA,
+    ES256,
+    ES256K,
+    ES384,
+}
+
+impl From<SigType> for &str {
+    fn from(value: SigType) -> Self {
+        match value {
+            SigType::EdDSA => "eddsa",
+            SigType::ES256 => "es256",
+            SigType::ES256K => "es256k",
+            SigType::ES384 => "es384",
+        }
+    }
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum EntryTag {
+    Encrypted(String, String),
+    Plaintext(String, String),
+}
+
+#[derive(Debug, Default, Clone, Builder)]
+pub struct Record {
```

Comment:
Is this structure inspired based on what askar's storage model looks like?

Reply:
Yes.
https://github.com/hyperledger/aries-askar/blob/main/askar-storage/src/entry.rs#L25-L42